Streamline Windows installer page

timsneath opened the pull request:
Simplify manual installation instructions for Windows by:

- Creating a streamlined instructions page for installing Swift 6 on Windows without a package manager
- Relocating the laborious instructions for older (5.9 and earlier) Swift releases to a separate page
- Updating the top-level Windows installer page to include links to the new pages

I've also reviewed the instructions for modern Swift builds and:
- Harmonized them with the instructions for package managers (e.g. don't require CMake Project Tools)
- Changed the dependencies for Python and Git to point to the source locations, rather than the vended versions in Visual Studio (since Python in particular is no longer vended)

Fixes #845

timsneath commented:
@swift-ci please test

compnerd commented on install/windows/_manual.md:
```diff
@@ -0,0 +1,34 @@
+
+You should also install the following dependencies:
+
+- [Python 3 (latest stable release)](https://www.python.org/downloads/windows/)
```
The latest version is not recommended. You should be using 3.10.x as that is what we build against and because of the way that LLDB uses Python, we need the version match.

timsneath replied:
Good catch -- thanks!

timsneath replied:
Good catch -- thanks! When does this change, though? If I put a static version in, won't it become quickly stale (the original

compnerd replied:
Yes, it does run the risk of becoming stale. Unfortunately, I don't think that there is a good solution here, unless you want to try to parse [build.ps1](https://github.com/swiftlang/swift/blob/main/utils/build.ps1).

This changes somewhat infrequently - I am trying to match the major/minor version that Apple has on their OS (i.e. 3.10). You can use the latest patch release for that version and it should be fine.

timsneath replied:
@compnerd Actually I have more questions here, if you'll indulge me. You said that build.ps1 shows the version that's used to build Swift -- but the [current version there](https://github.com/swiftlang/swift/blob/fb0b3fa90c72ae9de6a17d97097e706298ff2d02/utils/build.ps1#L130) is 3.9.10, not 3.10.x. Did I miss something?

And regardless of whether it's 3.9 or 3.10, what do we expect an ARM64 user to do? As far as I can tell, the first ARM64 version supported is [3.11.0](https://www.python.org/downloads/release/python-3110/), and I wasn't able to get AMD64 builds to successfully install in emulation mode.

compnerd replied:
Ah, sorry, I must've misremembered the version. For ARM64, we currently pull a version from nuget. This is the only way that we have to avoid a full build of python as a dependency.

As to the criticality of it - without the specific version being installed and on the path, lldb will not function. This extends to the REPL, though that is currently known to have a regression with 6.x. The missing functionality of lldb will extend to any IDE (e.g. VSCode).

Given that LLDB is part of the expected experience for Swift, I would consider it pretty critical.

I am reluctant to move the dependency independently of the other platforms (i.e. Apple). At the same time, even if we were to move the dependency independently, we wouldn't see that in practice until swift next. For the time being it would appear that we are stuck with this version.

compnerd commented on install/windows/_manual.md:
```diff
@@ -0,0 +1,34 @@
+
+Alternatively, you may prefer to install a [development snapshot](/install/windows/#development-snapshots) for access to features that are actively under development. 
+
+The toolchain on Windows is typically installed to `%SystemDrive%\Library\Developer\Toolchains`.
```
This path has now changed with 5.9 I think? We install to a per-user install location (`%LocalAppData%\Programs\Swift\Toolchains`).

compnerd commented on install/windows/manual/index.md:
```diff
@@ -0,0 +1,39 @@
+
+You should also install the following dependencies:
+
+- [Python 3 (latest stable release)](https://www.python.org/downloads/windows/)
```
Same as above

compnerd commented on install/windows/manual/index.md:
```diff
@@ -0,0 +1,39 @@
+
+Alternatively, you may prefer to install a [development snapshot](/install/windows/#development-snapshots) for access to features that are actively under development. 
+
+The toolchain on Windows is typically installed to `%SystemDrive%\Library\Developer\Toolchains`.
```
And here too

timsneath commented:
@compnerd Thanks for your feedback; please take another look and see if this addresses your comments. In particular, I've  added a footnote to include your explanation on which Python version should be selected. 

@swift-ci please test

compnerd reviewed (approved):
Perhaps use `archived` instead of `old`? But if that doesn't show up in the URL, I don't think that it really matters at all.

timsneath commented:
@swift-ci please test again. thanks for all you do behind the scenes, my little unsung hero 🦸

shahmishal commented:
@swift-ci please test

timsneath commented:
@swift-ci please test
